### ISSUE TITLE
What's the search method's expression

### ISSUE THREAD

**Issue by DAOStartup** [spring-projects/spring-data-rest#2175]
I know JPA has a way to construct the query by build a method's expression, but what is the data rest's construct way?
For example, I want to searh a keyword which was contained in filed1 and filed2. Should I name it with the name (maybe) file1OrField2ContainsWith(String keyword) ?. What is the correct way to build that method ?

**Comment by gregturn** [comment]
Simply put, Spring Data REST leverages the underlying repository's methods and entity types. 

For example, if you had an `EmployeeRepository` built off `Employee` and it contained some relationship to a `Manager` entity, and you then created `findByManagerName(String name)` in the `EmployeeRepository`, this finder would appear in Spring Data REST's `search` section of the API.

All the attributes shown on the endpoint are derived from the entity definitions, so there isn't really any form of "doing it the Data REST way."

If you are wanting to fashion a custom finder that operations on the entity type's `firstName` and `lastName` fields, then you can write something like `findByFirstNameContainsOrLastNameContainsAllIgnoreCase(String firstNamePartial, String lastNamePartial)`. From here you could plug in the one common value you are searching against, into both fields.

Basically, build things repository first, and then your Data REST end points will follow. If you wish to further explore this, I suggest posting your usage question on stackoverflow.com.
